### ISSUE TITLE
Discussion thread: How to handle seperate turnin lanes for a road with bus lanes?

### ISSUE THREAD

**Issue by kianzarrin** [CitiesSkylinesMods/TMPE#752]
separating turning lanes on roads with bus lanes is not to user satisfaction. on a 3 lane road bus lane becomes the dedicated right turn which is strange:
![Screenshot (629)](https://user-images.githubusercontent.com/26344691/75588013-7399e280-5a80-11ea-8e90-abfec6729039.png)

while fixing #567 I have provided back-end code that can support any different ways of calculating lanes. we only need to make a decision on one of these appraoches:
1. mix bus lanes and normal lanes together(what we are doing now) (screenshot 1 above)
2. calculate bus lanes and normal lanes separately (screenshot 2)
3. exclude bus lanes from calculations (screenshot 3)
4. depends on configuration whether or not cars can use bus lanes.
5. - alternate with phase

![Screenshot (630)](https://user-images.githubusercontent.com/26344691/75588003-6da40180-5a80-11ea-9dbd-9dc524c17cc6.png)
![Screenshot (631)](https://user-images.githubusercontent.com/26344691/75587994-67ae2080-5a80-11ea-836c-4abd6df544bc.png)

approach 1 or 4 can be justified because cars do use bus lanes  as mentioned in https://github.com/CitiesSkylinesMods/TMPE/issues/716#issuecomment-592252788 . but they are less willing to do so which is not very nice. if we add the option to allow cars to use bus lanes (without penalty) approach 1 would be even nicer.

approach 2 and 3 have no influence when there are 2 lanes (1 normal 1 bus). approach 3 looks weird on big roads. approach 2 seems more appropriate.

**Comment by kianzarrin** [comment]
There is also a bonus approach 6.: normal lanes get dedicated turns, bus lanes mix forward and turning lanes:
![Screenshot (634)](https://user-images.githubusercontent.com/26344691/75589156-c1174f00-5a82-11ea-8a15-5a019abd54f3.png)

it allows the inner bus lane to turn left and go forward which is not possible in any of the approaches 1-5 . is there any benefit in doing that?

In any case chaining the lane arrow of one bus line should not be too difficult for the user.

**Comment by aubergine10** [comment]
Personally I think 

* depends on configuration whether or not cars can use bus lanes, then:
    * calculate bus lanes and normal lanes separately (screenshot 2)

**Comment by aubergine10** [comment]
Hrm, rethinking what I said above.

Notably, bus lanes aren't just used by busses - they are used by service vehicles and taxis too. So they are not quite as empty as they would be if it was just busses.

We don't really want traffic in adjacent lane cutting across bus lane traffic if that traffic is going straight ahead.

In cases where there is a cross roads (or even more roads at junction), the edge lanes should be turning lanes (if there are enough lanes). If bus lane traffic wants to go straight ahead it should be in the central lanes ideally. Or there should be traffic lights so the bus lane traffic can be stopped while normal traffic cuts across it.

As there isn't really any way to convey all that to user, maybe the solition is that the 'turning lane separator' tool does nothing at such junctions and leave it to the user to manually configure their own accident hotspot.

If we take that approach generally, the inability to apply a quick setup tool or bulk applicator at a junction will be like a warning to the user that their roads are death traps.

**Comment by kianzarrin** [comment]
> We don't really want traffic in adjacent lane cutting across bus lane if its (bus lane) traffic is going straight ahead.

Isn't that what British  do?
![bus-lane-left-turn](https://user-images.githubusercontent.com/26344691/75597335-95559280-5a9d-11ea-97d6-c6b7ab1c01cd.jpg)

> EDIT: If user allows vehicles to use bus lane as turning lane, then bus lane could be ahead+turn, but adjacent normal lane should then be ahead-only.

So basically we should not separate turning lanes when there are bus lanes :) .

For the sake of #567 I added state machine to switch between approach 3 and 1. We can decide what do while fixing this issue.

**Comment by aubergine10** [comment]
> Isn't that what British do?

Yes, but technically the bus lane ends and busses give way to cars making a turn. (In reality, it's often the other way round; cars wait in their lane, holding up traffic, until bus is clear of the junction).
